api/servers_handler.go

```diff
@@ -0,0 +1,41 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/topfreegames/pitaya"
+)
+
+// ServersHandler handler
+type ServersHandler struct {
+	App *App
+}
+
+// NewServersHandler creates a new servers handler
+func NewServersHandler(a *App) *ServersHandler {
+	m := &ServersHandler{App: a}
+	return m
+}
+
+// ServeHTTP method
+func (s *ServersHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	param := strings.Replace(r.URL.Path, "/servers/", "", -1)
+	serversMap, err := pitaya.GetServersByType(param)
+
+	if err != nil {
+		w.Write([]byte("error: failed to retrieve servers by type"))
+		return
+	}
+
+	for _, server := range serversMap {
+		serverJSON, err := json.Marshal(server)
+		if err == nil {
```

Comment:
we usually make the check the other way around:
```
if err != nil {
  w.Write([]byte("error: failed to retrieve server details"))
}
```

also, we don't want to write an answer for each server, we want o build a json with all servers and return them all